service.py

```diff
@@ -125,12 +109,14 @@ def get_study_kf_id(self, study_id):
             return self.external_ids[study_id], self.version[study_id]
         while retry_count > 1:
             resp = requests.get(
-                self.api+'/studies?external_id='+study_id)
+                self.api+'/studies?external_id='+study_id,
+                timeout=self.context.get_remaining_time_in_millis()-14000)
             if resp.status_code != 500:
                 break
             else:
-                time.sleep(1)
                 retry_count = retry_count - 1
+        if resp.status_code != 200:
+            raise TimeoutException
         if resp.status_code == 200 and len(resp.json()['results']) == 1:
```

Comment:
`if resp.status_code == 200` not needed here now because it raises an exception otherwise